spec/models/target_spec.rb

```diff
@@ -32,74 +36,95 @@
     end
   end
 
+  describe 'default scope' do
+    before do
+      @first_target = create(:target, due_date: 2.days.from_now)
+      @second_target = create(:target, due_date: 3.days.from_now)
+    end
+
+    it 'orders by ascending due_date' do
+      expect(Target.all).to match_array [@first_target, @second_target]
+    end
+  end
+
+  describe 'associations' do
+    it { should belong_to(:conference) }
+    it { should belong_to(:campaign) }
+  end
+
   describe '#get_progress' do
     it 'returns zero if there are no registrations' do
-      conference = build(:conference)
-      target = build(:target, target_count: 10, unit: Target.units[:registrations])
-      conference.targets = [target]
-
-      expect(target.get_progress).to eq('0')
+      expect(registration_target.get_progress).to eq('0')
     end
 
     it 'returns 10 if there one registrations of 10' do
-      conference = create(:conference)
-      target = create(:target, target_count: 10, unit: Target.units[:registrations])
-      registration = create(:registration)
-
-      conference.targets = [target]
-      conference.registrations = [registration]
+      create(:registration, conference: registration_target.conference)
 
-      expect(target.get_progress).to eq('10')
+      expect(registration_target.get_progress).to eq('10')
     end
 
     it 'returns zero if there are no submissions' do
-      conference = create(:conference)
-      target = build(:target, target_count: 10, unit: Target.units[:submissions])
-      conference.targets = [target]
-
-      expect(target.get_progress).to eq('0')
+      expect(submission_target.get_progress).to eq('0')
     end
 
     it 'returns 10 if there one submissions of 10' do
-      conference = create(:conference)
-      target = create(:target, target_count: 10, unit: Target.units[:submissions])
-      event = create(:event)
-
-      conference.targets = [target]
-      conference.program.events = [event]
+      submission_target.conference.program.events = [event]
 
-      expect(target.get_progress).to eq('10')
+      expect(submission_target.get_progress).to eq('10')
     end
 
     it 'returns zero if there are no program minutes' do
-      conference = create(:conference)
-      target = build(:target, target_count: 10, unit: Target.units[:program_minutes])
-      conference.targets = [target]
-
-      expect(target.get_progress).to eq('0')
+      expect(program_minutes_target.get_progress).to eq('0')
     end
 
     it 'returns 10 if there are 30 program minutes of 300' do
-      conference = create(:conference)
-      target = create(:target, target_count: 300, unit: Target.units[:program_minutes])
-      event = create(:event)
+      program_minutes_target.conference.program.events = [event]
 
-      conference.targets = [target]
-      conference.program.events = [event]
+      expect(program_minutes_target.get_progress).to eq('10')
+    end
+  end
+
+  describe '#get_campaign' do
```

Comment:
While the progress is the most important value in this hash, the test should be checking the overal functionality of `get_campaign`: if the function actually returns a hash, which is called result and contains all the values it is supposed to.